lib/log/log.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package log
+
+import (
+	"io"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Logger defines a subset of the structured logging interface
+type Logger interface {
```

Comment:
This interface is the basis for re-implementation of loggers (i.e. fsm logger that'd like to look like `logrus.FieldLogger` but cannot due to `FieldLogger` not being totally generic).